nautilus/src/nautilus/compiler/backends/mlir/CMakeLists.txt

```diff
@@ -1,7 +1,29 @@
 
 if (ENABLE_MLIR_BACKEND)
-
-    include(${CMAKE_CURRENT_SOURCE_DIR}/../../../../../../cmake/AddMLIR.cmake)
+    find_package(MLIR REQUIRED CONFIG)
+    message(STATUS MLIR_INCLUDE_DIRS=${MLIR_INCLUDE_DIRS})
+    target_include_directories(nautilus SYSTEM PRIVATE
+            $<BUILD_INTERFACE:${MLIR_INCLUDE_DIRS}>
+            $<INSTALL_INTERFACE:include/nebulastream/>)
+    add_compile_definitions(MLIR_COMPILER)
+    get_property(mlir_libs GLOBAL PROPERTY MLIR_ALL_LIBS)
+    get_property(dialect_libs GLOBAL PROPERTY MLIR_DIALECT_LIBS)
+    get_property(conversion_libs GLOBAL PROPERTY MLIR_CONVERSION_LIBS)
+    get_property(extension_libs GLOBAL PROPERTY MLIR_EXTENSION_LIBS)
+    message(STATUS "Using MLIRConfig.cmake in: ${MLIR_DIR}")
+    message(STATUS "mlir_libs: ${mlir_libs}")
+    message(STATUS "dialect_libs: ${dialect_libs}")
+    message(STATUS "conversion_libs: ${conversion_libs}")
+    message(STATUS "extension_libs: ${extension_libs}")
+    target_link_libraries(nautilus
+            PUBLIC
```

Comment:
why not link private here?

Reply:
Good catch, I was experimenting if changing to public would influence what gets exported during install. It did not.

I'll change it back to private.